tests/ui/borrow_as_ptr.rs

```diff
@@ -47,3 +47,8 @@ fn implicit_cast() {
     // Do not lint references to temporaries
     core::ptr::eq(&0i32, &1i32);
 }
+
+fn issue_15141() {
```

Comment:
A small comment like `// don't lint casts to dyn trait pointers` would be nice.

Reply:
Done